### PR TITLE
Add AccessPolicy for master ViewSets and RedirectContentGuard

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #2723
fixes: #2725
fixes: #2726

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tests/functional/api/test_scoping.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    with alice:
+        aclist = content_guards_api_client.list()
+        assert aclist.count >= 2
+    with bob:
+        bclist = content_guards_api_client.list()
+        assert bclist.count == 0
+        gen_object_with_cleanup(redirect_contentguard_api_client, {"name": str(uuid.uuid4())})
+        gen_object_with_cleanup(rbac_contentguard_api_client, {"name": str(uuid.uuid4())})
+        bclist = content_guards_api_client.list()
+        assert bclist.count >= 0
```
Comment (ipanova) [inline pulpcore/tests/functional/api/test_scoping.py:58]: i think you want here 2+

**Review by ipanova — COMMENTED** [review]
looks great, left some minor comments.

File: pulpcore/tests/functional/__init__.py [hunk 2]
```diff
@@ -52,69 +51,54 @@ def pytest_configure(config):
...
-    yield value
-    print(f"Correlation-ID = {value}")
-
-
-@pytest.fixture
-def pulpcore_client(cid, bindings_cfg):
-    api_client = ApiClient(bindings_cfg)
-    api_client.default_headers["Correlation-ID"] = cid
-    return api_client
+@pytest.fixture(scope="session")
```
Comment (bmbouter) [inline pulpcore/tests/functional/__init__.py:54]: What motivated these changes? I think @mdellweg wanted these to have the Correlation-ID value for log purposes, and so he switched them to be function scope.

Comment (gerrod3) [reply pulpcore/tests/functional/__init__.py:54]: This is just a bad rebase, I need to fix this.

File: pulpcore/tests/functional/__init__.py [hunk 3]
```diff
@@ -52,69 +51,54 @@ def pytest_configure(config):
...
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def users_roles_api_client(pulpcore_client):
     return UsersRolesApi(pulpcore_client)
 
 
-@pytest.fixture
-def roles_api_client(pulpcore_client):
```
Comment (bmbouter) [inline pulpcore/tests/functional/__init__.py:100]: We still need this one right?

File: pulpcore/tests/functional/__init__.py (hunk 2 above)
Comment (bmbouter) [reply pulpcore/tests/functional/__init__.py:54]: Everything else in here is looking great!

**Review by ipanova — APPROVED** [review]
Thank you!

**Review by bmbouter — APPROVED** [review]
This looks great!

**Pull request merged**